Submitted By:            zippo zippo@oppiz.net
Date:                    2009-09-20
Initial Package Version: 1.2.2
Upstream Status:         Unknown
Origin:                  Gentoo Curtis Magyar http://bugs.gentoo.org/270414
Description: Fixes getline and fexecve with glibc 2.10
diff -puNr cdrdao-1.2.2.orig/scsilib/include/schily.h cdrdao-1.2.2/scsilib/include/schily.h
--- cdrdao-1.2.2.orig/scsilib/include/schily.h	2004-03-05 01:30:40.000000000 +0100
+++ cdrdao-1.2.2/scsilib/include/schily.h	2009-03-11 21:55:15.000000000 +0100
@@ -108,7 +108,7 @@ extern	int	fexecle __PR((const char *, F
 		/* 6th arg not const, fexecv forces av[ac] = NULL */
 extern	int	fexecv __PR((const char *, FILE *, FILE *, FILE *, int,
 							char **));
-extern	int	fexecve __PR((const char *, FILE *, FILE *, FILE *,
+extern	int	f_execve __PR((const char *, FILE *, FILE *, FILE *,
 					char * const *, char * const *));
 extern	int	fspawnv __PR((FILE *, FILE *, FILE *, int, char * const *));
 extern	int	fspawnl __PR((FILE *, FILE *, FILE *,
@@ -187,7 +187,7 @@ extern	char	*fillbytes __PR((void *, int
 extern	char	*findbytes __PR((const void *, int, char));
 extern	int	findline __PR((const char *, char, const char *,
 							int, char **, int));
-extern	int	getline __PR((char *, int));
+extern	int	get_line __PR((char *, int));
 extern	int	getstr __PR((char *, int));
 extern	int	breakline __PR((char *, char, char **, int));
 extern	int	getallargs __PR((int *, char * const**, const char *, ...));
diff -puNr cdrdao-1.2.2.orig/scsilib/libscg/scsitransp.c cdrdao-1.2.2/scsilib/libscg/scsitransp.c
--- cdrdao-1.2.2.orig/scsilib/libscg/scsitransp.c	2004-06-17 22:20:27.000000000 +0200
+++ cdrdao-1.2.2/scsilib/libscg/scsitransp.c	2009-03-11 21:54:24.000000000 +0100
@@ -323,7 +323,7 @@ scg_yes(msg)
 
 	js_printf("%s", msg);
 	flush();
-	if (getline(okbuf, sizeof (okbuf)) == EOF)
+	if (get_line(okbuf, sizeof (okbuf)) == EOF)
 		exit(EX_BAD);
 	if (streql(okbuf, "y") || streql(okbuf, "yes") ||
 	    streql(okbuf, "Y") || streql(okbuf, "YES"))
diff -puNr cdrdao-1.2.2.orig/scsilib/libschily/fexec.c cdrdao-1.2.2/scsilib/libschily/fexec.c
--- cdrdao-1.2.2.orig/scsilib/libschily/fexec.c	2004-06-06 13:50:24.000000000 +0200
+++ cdrdao-1.2.2/scsilib/libschily/fexec.c	2009-03-11 21:55:06.000000000 +0100
@@ -159,7 +159,7 @@ fexecle(name, in, out, err, va_alist)
 	} while (p != NULL);
 	va_end(args);
 
-	ret = fexecve(name, in, out, err, av, env);
+	ret = f_execve(name, in, out, err, av, env);
 	if (av != xav)
 		free(av);
 	return (ret);
@@ -173,11 +173,11 @@ fexecv(name, in, out, err, ac, av)
 	char *av[];
 {
 	av[ac] = NULL;			/*  force list to be null terminated */
-	return (fexecve(name, in, out, err, av, environ));
+	return (f_execve(name, in, out, err, av, environ));
 }
 
 EXPORT int
-fexecve(name, in, out, err, av, env)
+f_execve(name, in, out, err, av, env)
 	const char *name;
 	FILE *in, *out, *err;
 	char * const av[], * const env[];
diff -puNr cdrdao-1.2.2.orig/scsilib/libschily/stdio/fgetline.c cdrdao-1.2.2/scsilib/libschily/stdio/fgetline.c
--- cdrdao-1.2.2.orig/scsilib/libschily/stdio/fgetline.c	2004-08-08 13:02:05.000000000 +0200
+++ cdrdao-1.2.2/scsilib/libschily/stdio/fgetline.c	2009-03-11 21:54:04.000000000 +0100
@@ -64,7 +64,7 @@ fgetline(f, buf, len)
 }
 
 EXPORT int
-getline(buf, len)
+get_line(buf, len)
 	char	*buf;
 	int	len;
 {
